File: 908-middle-of-the-linked-list/middle-of-the-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        ListNode* curr = head;
        int size = 0;
        while(curr != nullptr)
        {
            size++;
            curr = curr->next;   
        }
        cout << size << endl;
        int halfList = ceil((double)size/2);
        curr = head;
        while(curr != nullptr && size != halfList)
        {
            size--;
            curr = curr->next;
        }
        return curr;
    }
};

File: 908-middle-of-the-linked-list/middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        fastNode = head
        slowNode = head
        while fastNode != None and fastNode.next != None:
            fastNode = fastNode.next.next
            slowNode = slowNode.next
        return slowNode
            